onchain/rollups/test/foundry/consensus/quorum/QuorumFactory.t.sol

```diff
@@ -0,0 +1,137 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+/// @title Quorum Factory Test
+pragma solidity ^0.8.8;
+
+import {Test} from "forge-std/Test.sol";
+import {QuorumFactory} from "contracts/consensus/quorum/QuorumFactory.sol";
+import {Quorum} from "contracts/consensus/quorum/Quorum.sol";
+import {IHistory} from "contracts/history/IHistory.sol";
+import {Vm} from "forge-std/Vm.sol";
+
+contract QuorumFactoryTest is Test {
+    QuorumFactory factory;
+
+    event QuorumCreated(Quorum quorum, IHistory history);
+    struct QuorumCreatedEventData {
+        Quorum quorum;
+        IHistory history;
+    }
+
+    function setUp() public {
+        factory = new QuorumFactory();
+    }
+
+    function testNewQuorum(uint256 _seed, IHistory _history) public {
+        (
+            address[] memory validators,
+            uint256[] memory shares
+        ) = generateAddrAndShares(_seed);
+
+        vm.recordLogs();
+
+        Quorum quorum = factory.newQuorum(validators, shares, _history);
+
+        testNewQuorumAux(validators, shares, _history, quorum);
+    }
+
+    function generateAddrAndShares(
+        uint256 _seed
+    )
+        internal
+        view
+        returns (address[] memory validators_, uint256[] memory shares_)
```

Comment:
I think it's simpler to be 1 function. There's no demand to generate solely the array of validators, or solely the array of shares. If we split it into 2 functions, the main function bodies of them would be the same, thus increasing code redundancy. 
As for limiting to `uint8`. I did intentionally choose a number greater than 255 in the test, because BitMap uses a new storage slot for a quorum greater than 255. Although in this factory tests, it doesn't have any effect because we don't submit claims or anything. Just to explain why I had in mind a quorum greater than 255. I think it's more flexity to not limit the quorum size to `uint8.max`

Reply:
Oh, good catch. I think I will update the quorum tests to encompass that then!

Reply:
But then we would have 2 function bodies that have basically the same code logic, which adds redundancy.

Reply:
Unless we want to have those individual functions in the `util/TestBase.sol`. Then it makes sense to modularize them. What do you think

Reply:
Yes, let's move them to `TestBase`:
- `generateAddresses(uint256 n) -> address[]`
- `generateArithmeticSequence(uint256 a0, uint256 d, uint256 n) -> uint256[]`

Reply:
In fact, let me do that on `feature/quorum`. I can use these for the Quorum tests as well!

Reply:
Yep. I thought the same too :)